modules/dataquery/php/endpoints/queries/query/run.class.inc

```diff
@@ -55,48 +60,49 @@ class Run extends \LORIS\Http\Endpoint
      */
     public function runQuery(\User $user, int $queryID) : ResponseInterface
     {
-        try {
-            $query = new \LORIS\dataquery\Query($this->loris, $queryID);
+        $query = new \LORIS\dataquery\Query($this->loris, $queryID);
 
-            $queryRun = $query->newRun($user);
-            $queryRun->insertCandidates($user);
-            $data = $queryRun->getQueryDataProvisioner();
+        $db = $this->loris->getDatabaseConnection();
+        // Put the query run in a transaction to make it atomic
+        $queryRun = $query->newRun($user);
+        $db->beginTransaction();
+        $data = $queryRun->getQueryDataProvisioner()->execute($user);
 
-            $table  = (new \LORIS\Data\Table())
-                ->withDataFrom($data);
-            $rows   = $table->getRows($user);
-            $stream = new \LORIS\Http\DataIteratorBinaryStream($rows);
-            return (new \LORIS\Http\Response())
-                ->withHeader("Content-Type", "text/plain")
-                //->withHeader("Content-Type", "application/octet-stream")
-                ->withBody($stream);
-        } catch (\LorisException $e) {
-            return new \LORIS\Http\Response\JSON\NotImplemented($e->getMessage());
+        $values = [];
 
-        }
-    }
+        set_time_limit(300);
```

Comment:
Should this be configurable?

Reply:
Maybe?

I think it can just be removed. Our documentation says to set `max_execution_time` to 10800 to support large files which is the php config setting that php would fall back on.

https://acesloris.readthedocs.io/en/latest/docs/wiki/00_SERVER_INSTALL_AND_CONFIGURATION/02_Website_Configuration/Handling_Large_File_Uploads.html